engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -189,14 +190,43 @@ public final class QueryLanguageParser extends GenericVisitorAdapter<Class<?>, Q
      * @param variableTypeArguments A map of the names of scope variables to their type arguments
      * @throws QueryLanguageParseException If any exception or error is encountered
      */
-    public QueryLanguageParser(String expression,
+    @TestUseOnly
+    QueryLanguageParser(
+            String expression,
             Collection<Package> packageImports,
             Collection<Class<?>> classImports,
             Collection<Class<?>> staticImports,
             Map<String, Class<?>> variables,
             Map<String, Class<?>[]> variableTypeArguments) throws QueryLanguageParseException {
         this(expression, packageImports, classImports, staticImports, variables,
-                variableTypeArguments, true);
+                variableTypeArguments, null, true);
+    }
+
+    /**
+     * Create a QueryLanguageParser and parse the given {@code expression}. After construction, the
+     * {@link QueryLanguageParser.Result result} of parsing the {@code expression} is available with the
+     * {@link #getResult()}} method.
+     *
+     * @param expression The query language expression to parse
+     * @param packageImports Wildcard package imports
+     * @param classImports Individual class imports
+     * @param staticImports Wildcard static imports. All static variables and methods for the given classes are
+     *        imported.
+     * @param variables A map of the names of scope variables to their types
+     * @param variableTypeArguments A map of the names of scope variables to their type arguments
+     * @param possibleParams A mutable map of the names of query scope variables to their values
```

Comment:
Don't quite like the name `possibleParams`, I think `queryScopeVariables` might be easier for readers.

Reply:
This is a good point, and I endorse it whole-heartedly.